x-pack/plugins/lens/public/editor_frame_service/editor_frame/suggestion_panel.tsx

```diff
@@ -135,10 +141,13 @@ const SuggestionPreview = ({
           data-test-subj="lnsSuggestion"
           onClick={onSelect}
         >
+          {preview.expression && currentExpression !== renderedExpression && (
+            <EuiProgress size="xs" color="accent" position="absolute" />
+          )}
```

Comment:
I agree that this approach works, but does it need to be Lens-specific? What if we supported a debouncing parameter on the `ExpressionRenderer` component which would handle this automatically? The main reason I'm suggesting this is that the `ExpressionRenderer` component is already responsible for showing both an `EuiProgress` and `<EuiLoadingChart size="m" />`, and it might be easy to share the logic.

Reply:
@elastic/kibana-app-arch what do you think about having a “debounce” parameter on the expression renderer component basically replicating this logic one layer deeper? The use case here is to not re-render the suggestions too often while the user interacts with the config, but still show an indication the current state will be updated

Reply:
i don't see a problem with introducing debounce parameter to expression renderer component

Reply:
Thanks Peter, I will move the logic over on this PR